### ISSUE TITLE
[BUG] Smartspacer at glance provider limited to 5 pages/targets

### ISSUE THREAD

**Issue by pogman-code** [Deletescape-Media/Lawnchair#4328]
### Describe the bug

If Smartspacer has more than 5 target, only the first 5 are show in the "At a Glance" widget.

### Steps to reproduce

1. Install Smartspacer
2. Add more than 5 targets to Smartspacer
3. Go to Lawnchair Settings
4. Click on At A Glance
5. Change provider to Smartspacer
6. Go back to home screen
7. Only the 5 first smartspacer target are shown

### Expected behavior

All Smartspacer targets should be show in "At a Glance" widget.

### Device information

- Device: Xiaomi Mi 10 Lite 5G (Monet)
- OS: LineageOS 19.1
- App version: 14 Beta 2

**Comment by MrSluffy** [comment]
Please report it in smart spacer repo, lawnchair does not control the smart spacer target other than displaying it

**Comment by KieronQuinn** [comment]
> Please report it in smart spacer repo, lawnchair does not control the smart spacer target other than displaying it

_Technically_ it does, although it's currently hard coded to 5 pages as it's using the View which is designed to be drop in and not be able to override that: https://github.com/KieronQuinn/Smartspacer/blob/962beac6efba350a594eb11bef306e1a11fa4d27/sdk-client/src/main/java/com/kieronquinn/app/smartspacer/sdk/client/views/BcSmartspaceView.kt#L44

If I were to add an override option, Lawnchair would need to add a UI element to configure that. Alternatively, you could just use the override to set it to MAX_INT, which is what the unlimited option would do anyway.

What do you think?

**Comment by KieronQuinn** [comment]
I've just released 1.0.11 of the client SDK, which now exposes the SmartspaceConfig for BcSmartspaceView. 

If you wish to just disable the limit, simply override `config` in [SmartspacerView](https://github.com/LawnchairLauncher/lawnchair/blob/14-dev/lawnchair/src/app/lawnchair/smartspace/SmartspacerView.kt):

```
override val config = SmartspaceConfig(Int.MAX_VALUE, UiSurface.HOMESCREEN, context.packageName)
```

If you wish to add limit disabling as an option, you'll need to add further code however.

**Comment by SuperDragonXD** [comment]
I've added an option to override the maximum limit up to 15. I don't think it's necessary for us to have an infinite amount of pages; if one is interested in that however, I can add an option.
